### PR TITLE
Improve PID file error handling

### PR DESCRIPTION
**Pull request by hebasto**
Digging into #15240 the lack of the proper logging has been discovered.
Fixed by this PR.

UPDATE (inspired by @laanwj's [comment](https://github.com/bitcoin/bitcoin/pull/15278#discussion_r252641810)):
Not being able to create the PID file is fatal now.

Output of `bitcoind`:

```
$ src/bitcoind -pid=/run/bitcoind/bitcoind.pid
2019-02-01T23:20:10Z Bitcoin Core version v0.17.99.0-561e375c7 (release build)
2019-02-01T23:20:10Z Assuming ancestors of block 0000000000000037a8cd3e06cd5edbfe9dd1dbcc5dacab279376ef7cfc2b4c75 have valid signatures.
2019-02-01T23:20:10Z Setting nMinimumChainWork=00000000000000000000000000000000000000000000007dbe94253893cbd463
2019-02-01T23:20:10Z Using the 'sse4(1way),sse41(4way),avx2(8way)' SHA256 implementation
2019-02-01T23:20:10Z Using RdRand as an additional entropy source
2019-02-01T23:20:11Z Error: Unable to create the PID file '/run/bitcoind/bitcoind.pid': No such file or directory
Error: Unable to create the PID file '/run/bitcoind/bitcoind.pid': No such file or directory
2019-02-01T23:20:11Z Shutdown: In progress...
2019-02-01T23:20:11Z Shutdown: Unable to remove PID file: File does not exist
2019-02-01T23:20:11Z Shutdown: done
```

Output of `bitcoin-qt`:
screenshot from 2019-02-02 01-19-05

**Notes for reviewers**
1. `CreatePidFile()` has been moved from `util/system.cpp` to `init.cpp` for the following reasons:
- to get the ability to use `InitError()`
- now `init.cpp` contains code of both creating PID file and removing it

2. Regarding 0.18 release process: this PR modifies 1 string and introduces 2 new ones.

### CODE REVIEW

File: src/util/system.cpp [hunk 1]
```diff
@@ -967,10 +968,11 @@ fs::path GetPidFile()
 void CreatePidFile(const fs::path &path, pid_t pid)
```
Comment (laanwj) [inline src/util/system.cpp]: I don't quite understand why this doesn't return an error status, and that not being able to create the PID file is not fatal.

**Comment by laanwj** [comment]
Concept ACK, thanks for adding more relevant logging for diagnostics.

**Comment by practicalswift** [comment]
Concept ACK. I suggest adding the `NODISCARD` annotations to `fs::remove` and `CreatePidFile` (after making it return an error status as suggested by @laanwj).

**Comment by laanwj** [comment]
utACK 561e375c73a37934fe77a519762d81edf7a3325c
Thanks for fixing this

File: src/init.cpp [hunk 2]
```diff
@@ -1190,12 +1194,29 @@ bool AppInitLockDataDirectory()
     return true;
 }
 
+#ifndef WIN32
+NODISCARD static bool CreatePidFile()
```
Comment (laanwj) [inline src/init.cpp]: If you're moving this function anyway, why not move `GetPidFile` as well, it's only used from `init.cpp` too

**Comment by laanwj** [comment]
Tested:

- Removing the PId file while running, this results in the following warning in the log:
```
2019-02-13T22:07:02Z Shutdown: Unable to remove PID file: File does not exist
```
- After making the pid file unaccessible by chowning it to root:
```
2019-02-13T22:08:36Z Error: Unable to create the PID file '/.../.bitcoin/regtest/bitcoind.pid': Permission denied
Error: Unable to create the PID file '/home/orion/.bitcoin/regtest/bitcoind.pid': Permission deniez
```

Looks good to me.

**Comment by hebasto** [comment]
@laanwj's https://github.com/bitcoin/bitcoin/pull/15278#discussion_r256606572 has been addressed.

File: src/init.cpp (hunk 2 above)
Comment (hebasto) [reply src/init.cpp]: Fixed.

**Comment by laanwj** [comment]
utACK 3782075a5fd4ad0c15a6119e8cdaf136898f679e

**Pull request merged**